site/app/controllers/admin/UsersController.php

```diff
@@ -411,7 +411,19 @@ public function updateRegistrationSections() {
             if (User::validateUserData('registration_section', $_POST['delete_reg_section'])) {
                 // DELETE trigger function in master DB will catch integrity violation exceptions (such as FK violations when users/graders are still enrolled in section).
                 // $num_del_sections indicates how many DELETEs were performed.  0 DELETEs means either the section didn't exist or there are users still enrolled.
-                $num_del_sections = $this->core->getQueries()->deleteRegistrationSection($_POST['delete_reg_section']);
+                $this->core->getConfig()->getCoursePath() . '/uploads/course_materials_file_data.json';
+                $json = file_get_contents($fp);
+                $jsonArray = json_decode($json, true);
```

Comment:
Looks like you've missed a `$fp = ` on line 414, however you can write the three lines as:
`$json = FileUtils::readJsonFile(FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), 'uploads', 'course_materials_file_data.json');`